src/stream_processor/flb_sp.c

```diff
@@ -526,6 +523,319 @@ struct flb_sp *flb_sp_create(struct flb_config *config)
+                        break;
+                    case FLB_EXP_FLOAT:
+                        result->val.boolean = (left->val.f64 == right->val.f64);
+                        break;
+                    case FLB_EXP_STRING:
+                        flb_info("string operation: %s %s", left->val.string, right->val.string);
+                        if (flb_sds_len(left->val.string) != flb_sds_len(right->val.string))
+                            result->val.boolean = false;
+                        else if (strncmp(left->val.string, right->val.string, flb_sds_len(left->val.string)) != 0)
+                            result->val.boolean = false;
+                        else
+                            result->val.boolean = true;
+                        break;
+                    default:
+                        result->val.boolean = false;
+                        break;
+                }
+            } else
+                result->val.boolean = false;
+            break;
+        case FLB_EXP_LT:
+            if (left->type == right->type)
+            {   switch(left->type){
+                    case FLB_EXP_INT:
+                        result->val.boolean = (left->val.i64 < right->val.i64);
+                        break;
+                    case FLB_EXP_FLOAT:
+                        result->val.boolean = (left->val.f64 < right->val.f64);
+                        break;
+                    case FLB_EXP_STRING:
+                        if (strncmp(left->val.string, right->val.string, flb_sds_len(left->val.string)) < 0)
+                            result->val.boolean = true;
+                        else
+                            result->val.boolean = false;
+                        break;
+                    default:
+                        result->val.boolean = false;
+                        break;
+                }
+            } else
+                result->val.boolean = false;
+            break;
+        case FLB_EXP_LTE:
+            if (left->type == right->type)
+            {   switch(left->type){
+                    case FLB_EXP_INT:
+                        result->val.boolean = (left->val.i64 <= right->val.i64);
+                        break;
+                    case FLB_EXP_FLOAT:
+                        result->val.boolean = (left->val.f64 <= right->val.f64);
+                        break;
+                    case FLB_EXP_STRING:
+                        if (strncmp(left->val.string, right->val.string, flb_sds_len(left->val.string)) <= 0)
+                            result->val.boolean = true;
+                        else
+                            result->val.boolean = false;
+                        break;
+                    default:
+                        result->val.boolean = false;
+                        break;
+                }
+            } else
+                result->val.boolean = false;
+            break;
+        case FLB_EXP_GT:
+            if (left->type == right->type)
```

Comment:
fix brace position: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions